mlflow/pytorch/_lightning_autolog.py

```diff
@@ -0,0 +1,460 @@
+            for p in ["monitor", "mode", "patience", "min_delta", "stopped_epoch"]
+            if hasattr(early_stop_callback, p)
+        },
+    )
+
+
+def _log_early_stop_metrics(early_stop_callback, client, run_id):
+    """
+    Logs early stopping behavior results (e.g. stopped epoch) as metrics to MLflow.
+
+    :param early_stop_callback: The early stopping callback instance used during training.
+    :param client: An `MlflowAutologgingQueueingClient` instance used for MLflow logging.
+    :param run_id: The ID of the MLflow Run to which to log configuration parameters.
+    """
+    if early_stop_callback.stopped_epoch == 0:
+        return
+
+    metrics = {
+        "stopped_epoch": early_stop_callback.stopped_epoch,
+        "restored_epoch": early_stop_callback.stopped_epoch - max(1, early_stop_callback.patience),
+    }
+
+    if hasattr(early_stop_callback, "best_score"):
+        metrics["best_score"] = float(early_stop_callback.best_score)
+
+    if hasattr(early_stop_callback, "wait_count"):
+        metrics["wait_count"] = early_stop_callback.wait_count
+
+    client.log_metrics(run_id, metrics)
+
+
+def patched_fit(original, self, *args, **kwargs):
+    """
+    A patched implementation of `pytorch_lightning.Trainer.fit` which enables logging the
+    following parameters, metrics and artifacts:
+
+    - Training epochs
+    - Optimizer parameters
+    - `EarlyStoppingCallback`_ parameters
+    - Metrics stored in `trainer.callback_metrics`
+    - Model checkpoints
+    - Trained model
+
+    .. _EarlyStoppingCallback:
+        https://pytorch-lightning.readthedocs.io/en/latest/early_stopping.html
+    """
+    global IN_FIT
+
+    import torch
+
+    def _get_version(pkg_name):
+        if sys.version_info < (3, 8):
+            import pkg_resources
+
+            return Version(pkg_resources.get_distribution(pkg_name).version)
+        from importlib.metadata import version
+
+        return Version(version(pkg_name))
+
+    if not MIN_REQ_VERSION <= _pl_version <= MAX_REQ_VERSION:
+        warnings.warn(
+            (
+                "Autologging known to be compatible with pytorch-lightning versions between "
+                "%s and %s and may not succeed with packages "
+                "outside this range."
+            )
+            % (MIN_REQ_VERSION, MAX_REQ_VERSION)
+        )
+
+    if _pl_version < Version("1.2.0") and Version(torch.__version__) >= Version("1.11.0"):
+        warnings.warn(
+            (
+                "Autologging for lightning versions <1.2.0 is only known to work with pytorch "
+                "versions <1.11.0 and may not succeed with packages outside this range."
+            )
+        )
+
+    if _get_version("protobuf") >= Version("4.0.0"):
+        warnings.warn(
+            (
+                "Autologging for lightning is only supported with protobuf version < 4.0.0 "
+                "and may not succeed with packages outside this range."
+            )
+        )
```

Comment:
```suggestion
```

We don't need to check the torch or protobuf versions.

Reply:
these checks are migrated from `ml-package-versions.yml`, I'm wondering where they should go instead if not here.

Reply:
We can just remove them. For pytorch-lightning autologging, we can just check the version of pytorch-lightning.